File: C++/프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> dot) {
    // y축 처리
    int answer = dot[1] > 0? 1 : 3;
    
    // x축 처리
    // x * y가 양수 이면 1, 3 분면
    // x * y가 음수 이면 2, 4 분면
    answer += dot[0] * dot[1] > 0 ? 0 : 1;
    return answer;
}

File: Go/프로그래머스/0/120841. 점의 위치 구하기/점의 위치 구하기.go
func solution(dot []int) int {
    var res int
    
    // 1,2 분면과 3,4 분면으로 분리
    if dot[1] > 0 {
        res = 1
    } else {
        res = 3
    }
    
    // x * y가 양수면 1,3분면, 음수면 2,4분면
    if dot[1] * dot[0] < 0 {
        res++
    }
    return res
}